src/Cohesity.Powershell/Scripts/Restore/Restore-CohesityRemoteFile.ps1

```diff
@@ -96,154 +96,153 @@ function Restore-CohesityRemoteFile {
+            $payload = @{
+                filenames        = @($FileNames)
+                name             = $TaskName
+                params           = @{
+                    targetEntity             = $targetEntity
+                    targetEntityParentSource = $targetEntityParentSource
+                    targetEntityCredentials  = @{
+                        username = $targetUserName
+                        password = $targetPassword
                     }
-                    sourceObjectInfo = @{
-                        jobId          = $sourceVMDetails.vmDocument.objectId.jobId
-                        jobInstanceId  = $JobRunId
-                        startTimeUsecs = $StartTime
-                        parentSource   = $parentSource
-                        entity         = $sourceObjectEntity
-                        jobUid         = $sourceVMDetails.vmDocument.objectId.jobUid
+                    restoreFilesPreferences  = @{
+                        restoreToOriginalPaths        = $restoreToOriginalPaths
+                        overrideOriginals             = -not ($DoNotOverwrite.IsPresent)
+                        preserveTimestamps            = $true
+                        preserveAcls                  = $true
+                        preserveAttributes            = -not ($DoNotPreserveAttributes.IsPresent)
+                        continueOnError               = $ContinueOnError.IsPresent
+                        alternateRestoreBaseDirectory = $NewBaseDirectory
                     }
+                    targetHostType           = $TARGET_HOST_TYPE
+                    useExistingAgent         = $false
                 }
-                $url = '/irisservices/api/v1/restoreFiles'
-                $payloadJson = $payload | ConvertTo-Json -Depth 100
+                sourceObjectInfo = @{
+                    jobId          = $sourceVMDetails.vmDocument.objectId.jobId
+                    jobInstanceId  = $JobRunId
+                    startTimeUsecs = $StartTime
+                    parentSource   = $parentSource
+                    entity         = $sourceObjectEntity
+                    jobUid         = $sourceVMDetails.vmDocument.objectId.jobUid
+                }
+            }
 
-                $resp = Invoke-RestApi -Method 'Post' -Uri $url -Body $payloadJson
-                if ($Global:CohesityAPIStatus.StatusCode -eq 200) {
-                    $taskId = $resp.restoreTask.performRestoreTaskState.base.taskId
-                    if ($taskId) {
-                        $resp = Get-CohesityRestoreTask -Ids $taskId
-                        $resp
-                    }
-                    else {
-                        $resp
-                    }
+            $version = $sourceVMDetails.vmDocument.versions | Where-Object {$_.instanceId.jobInstanceId -eq $JobRunId}
+            if(($version.replicaInfo.replicaVec | Sort-Object -Property {$_.target.type})[0].target.type -eq 3){
```

Comment:
The cmdlet assumes restore will be only via archival , we should support replicated cluster later so better to add a flag ?